codebase/src/java/disco/org/openlvc/disco/application/EntityStateStore.java

```diff
@@ -52,6 +56,8 @@ protected EntityStateStore( PduStore store )
 	{
 		this.byId = new ConcurrentHashMap<>();
 		this.byMarking = new ConcurrentHashMap<>();
+
+		this.isDrEnabled = store.app.getConfiguration()::getDeadReckoningEnabled;
```

Comment:
Very minor note, but I'd be inclined to make two small changes:
 - rename `store` to `parentStore` (as that makes the relationship between the down-chain EntityStateStore and the PduStore clear in the name (I told you it was minor)
 - I'd avoid the Boolean Supplier. Just store the parent and query it live. Generally, unless there is a great reason, in cases like this where we are creating connective tissue between components, I'd go for the whole object, rather than just one small cross-section of it (if that makes sense). It makes the connection more explicit.